### PR TITLE
#1048 Work Package Edit Autocomplete

### LINKED ISSUE #1048: Work Package Edit - Make Blocked By a multiple autcomplete
### Description

Make the blocked by text field in the work package edit be a mulitple autocomplete where the options are only other work packages in the same project.

### Acceptance Criteria

Successfully implements a multiple autocomplete with the correct options. 

### Proposed Solution

See Task list modal for an example of a multiple auto complete, or member editing view on the teams page. You will want to replace the blocked by text field in the work package edit form to be a multiple autocomplete. 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Made Autocomplete for blockers in edit work package

## Screenshots

Screen Shot 2023-06-24 at 11 35 11 PM
Screen Shot 2023-06-24 at 11 35 39 PM
Screen Shot 2023-06-24 at 11 38 39 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1048  (issue #)

### CODE REVIEW

**Comment by Peyton-McKee** [comment]
Removed FormLabel
Screen Shot 2023-06-28 at 12 55 04 PM

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx [hunk 1]
```diff
@@ -156,6 +165,7 @@ const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ wor
 
   const projectWbsString: string = wbsPipe({ ...workPackage.wbsNum, workPackageNumber: 0 });
 
+  console.log(blockedByOptions);
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:168]: remove

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx [hunk 2]
```diff
@@ -189,21 +199,33 @@ const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ wor
...
-          + ADD NEW BLOCKER
-        </Button>
+        <FormControl fullWidth>
+          <Controller
+            name="blockedBy"
+            control={control}
+            render={({ field: { onChange, value: formValue } }) => (
+              <Autocomplete
+                isOptionEqualToValue={(option, value) => {
+                  console.log(option, value);
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:209]: remove

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx [hunk 3]
```diff
@@ -105,15 +99,32 @@ const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ wor
...
+  if (projectIsError) {
+    return <ErrorPage message={projectError?.message} />;
+  }
+
+  const blockedByToAutocompleteOption = (workPackage: WorkPackage) => {
+    return { id: wbsPipe(workPackage.wbsNum), label: `${wbsPipe(workPackage.wbsNum)} - ${workPackage.name}` };
+  };
+
+  const blockedByOptions =
+    project.workPackages.filter((wp) => wp.id !== workPackage.id).map(blockedByToAutocompleteOption) || [];
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:125]: wps can only be blocked by wps in the same project?

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx [hunk 4]
```diff
@@ -189,21 +199,33 @@ const WorkPackageEditContainer: React.FC<WorkPackageEditContainerProps> = ({ wor
...
+                  console.log(option, value);
+                  return option.id === value.id;
+                }}
+                filterSelectedOptions
+                multiple
+                options={blockedByOptions}
+                getOptionLabel={(option) => option.label}
+                onChange={(_, value) => onChange(value.map((v) => v.id))}
+                value={formValue.map((v: string) => {
+                  let change = blockedByOptions.find((o) => o.id === v);
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:218]: no reason for this to be let, you can change it to const

**Review by RChandler234 — CHANGES_REQUESTED** [review]
this looks awesome, so much better!

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx (hunk 3 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:125]: yeah lol

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx (hunk 4 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditContainer.tsx:218]: ah this was when i messing with the weird type stuff

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**